### ISSUE TITLE
Move cropper-modal inside cropper-canvas?

### ISSUE THREAD

**Issue by davidhaimson** [fengyuanchen/cropper#286]
The cropper plugin is a remarkable piece of software. I am very grateful you have built it and shared it with the world. We are using version v0.9.2.

We are using it somewhat differently than it is commonly used, I suspect. We are not using zoom, and the crop box does not move. Instead, we initialize the canvas larger than the crop box, and make the crop box full browser window width and fixed height. Zoom is not enabled. The user crops by dragging the image within the crop box. Our options are as follows:

    {
        autoCrop: true,
        autoCropArea: 1.0,
        background: false,
        checkImageOrigin: false,
        dragCrop: false,
        guides: false,
        highlight: false,
        modal: true,
        movable: false,
        mouseWheelZoom: false,
        resizable: false,
        responsive: false,
        strict: true,
        touchDragZoom: false,
        zoomable: false
    }

Our designers wanted the parts of the image outside the crop box visible while dragging. I attach screen shots of what this looks like.

Before drag:
![cropper-before-drag](https://cloud.githubusercontent.com/assets/6216954/7436820/74a34ae0-f00e-11e4-835d-54d1c3a4b063.png)

During drag:
![cropper-during-drag](https://cloud.githubusercontent.com/assets/6216954/7436825/89b01a44-f00e-11e4-8e8b-49eb9f75d84d.png)

In order to make this work, I had to rearrange some of the internal cropper elements. In my built.cropper handler, I do the following:

    var cropper = $('.cropper-container');
    var cropperModal = $('.cropper-modal', cropper);
    var cropperCanvas = $('.cropper-canvas', cropper);
    cropperModal.detach().prependTo(cropperCanvas);

Then I update the CanvasData as described, and finally update the CropBoxData.

I've also overridden the CSS for these components (.scss format):

    // override styling of cropper itself
    .cropper-container {
        position: relative;

        .cropper-canvas {
            opacity: 0.5;

            .cropper-modal {
                background-color: $bulb-light-gray;
            }
        }

        .cropper-crop-box .cropper-view-box {
            outline: 6px solid rgba($bulb-medium-gray, 0.9);
            outline-offset: -6px;
        }
    }

    &.bulb-image-editor-is-moving {
        overflow: visible !important;

        .cropper-container {
            overflow: visible !important;
        }
    }

Rearranging the cropper elements has caused no problems for the way we are using it. I have also tried it with configurations more like the defaults, with no problems.

But I am uncomfortable performing surgery on somebody else's code. Therefore, I'd appreciate it if you would do one of the following things for us:

1. If this change breaks some invariant of your code, please suggest an alternative.

2. If it works for you, consider placing the cropper-modal inside the cropper-canvas in the cropper when you generate them, making this a supported configuration.

Thanks very much,
David Haimson

**Comment by fengyuanchen** [comment]
Why do you fork one and build it as what you want?

**Comment by davidhaimson** [comment]
I should think the answer to that would be obvious: The change I've requested would cost you very little (or so I imagine). Taking on the maintenance of a branch of this fairly complex project would cost me a lot of time and effort.

Would you accept a pull request for this change?

Thanks,
David Haimson

**Comment by fengyuanchen** [comment]
No, thanks, I don't want to move it as it unnecessary for most people.

**Comment by davidhaimson** [comment]
As you wish.
